state-chain/pallets/cf-pools/src/lib.rs

```diff
@@ -1422,7 +1357,149 @@ impl<T: Config> Pallet<T> {
+			T::SafeMode::get().limit_order_update_enabled,
+			Error::<T>::UpdatingLimitOrdersDisabled
+		);
+		Self::try_mutate_order(lp, base_asset, quote_asset, |asset_pair, pool| {
+			let tick = match (
+				pool.limit_orders_cache[side.to_sold_pair()]
+					.get(lp)
+					.and_then(|limit_orders| limit_orders.get(&id))
+					.cloned(),
+				option_tick,
+			) {
+				(None, None) => Err(Error::<T>::UnspecifiedOrderPrice),
+				(None, Some(tick)) => Ok(tick),
+				(Some(previous_tick), option_new_tick) => {
+					Self::inner_update_limit_order_at_tick(
+						pool,
+						lp,
+						asset_pair,
+						side,
+						id,
+						previous_tick,
+						IncreaseOrDecrease::Decrease(Amount::MAX),
+						NoOpStatus::Error,
+					)?;
+
+					Ok(option_new_tick.unwrap_or(previous_tick))
+				},
+			}?;
+			Self::inner_update_limit_order_at_tick(
+				pool,
+				lp,
+				asset_pair,
+				side,
+				id,
+				tick,
+				IncreaseOrDecrease::Increase(sell_amount.into()),
+				NoOpStatus::Allow,
+			)?;
+
+			Ok(())
+		})
+	}
+
+	/// Updates limit order closing the previous one if necessary (in case of tick change)
 	fn inner_update_limit_order(
+		lp: &T::AccountId,
+		base_asset: Asset,
+		quote_asset: Asset,
+		side: Side,
+		id: OrderId,
+		option_tick: Option<Tick>,
+		amount_change: IncreaseOrDecrease<AssetAmount>,
+	) -> DispatchResult {
+		Self::try_mutate_order(lp, base_asset, quote_asset, |asset_pair, pool| {
+			let tick = match (
+				pool.limit_orders_cache[side.to_sold_pair()]
+					.get(lp)
+					.and_then(|limit_orders| limit_orders.get(&id))
+					.cloned(),
+				option_tick,
+			) {
+				(None, None) => Err(Error::<T>::UnspecifiedOrderPrice),
+				(None, Some(tick)) | (Some(tick), None) => Ok(tick),
+				(Some(previous_tick), Some(new_tick)) => {
+					if previous_tick != new_tick {
+						let withdrawn_asset_amount = Self::inner_update_limit_order_at_tick(
+							pool,
+							lp,
+							asset_pair,
+							side,
+							id,
+							previous_tick,
+							IncreaseOrDecrease::Decrease(Amount::MAX),
+							NoOpStatus::Error,
+						)?;
+						Self::inner_update_limit_order_at_tick(
+							pool,
+							lp,
+							asset_pair,
+							side,
+							id,
+							new_tick,
+							IncreaseOrDecrease::Increase(withdrawn_asset_amount.into()),
+							NoOpStatus::Allow,
+						)?;
+					}
+
+					Ok(new_tick)
+				},
+			}?;
+			Self::inner_update_limit_order_at_tick(
+				pool,
+				lp,
+				asset_pair,
+				side,
+				id,
+				tick,
+				amount_change.map(|amount| amount.into()),
+				NoOpStatus::Error,
+			)?;
+
+			Ok(())
+		})
+	}
+
+	fn sweep_limit_order(
+		pool: &mut Pool<T>,
+		lp: &T::AccountId,
+		asset_pair: &AssetPair,
+		side: Side,
+		id: OrderId,
+		tick: Tick,
+	) -> DispatchResult {
+		let sold_amount_change = Self::inner_update_limit_order_at_tick(
+			pool,
+			lp,
+			asset_pair,
+			side,
+			id,
+			tick,
+			IncreaseOrDecrease::Decrease(Default::default()),
+			NoOpStatus::Error,
+		)?;
+
+		if sold_amount_change != 0 {
```

Comment:
Terminology not from your PR ,but this is the sold amount remaining? I was confused thinking it meant "change" as in delta... a comment somewhere about what "sold_amount_change" is - probably on the inner most function.

Reply:
It is whether we decreased or increased the amount we "offer"/"sell". Because we requested 0 change (since we are only sweeping), we expect 0 change as the result. I think in the context of `inner_update_limit_order_at_tick` is it actually pretty clear. I will add a comment here instead.